src/js/stores/AppFormStore.js

```diff
@@ -79,10 +84,15 @@ function getTransformedField(fieldId, value) {
 }
 
 function rebuildModelFromFields(app, fields, fieldId) {
-  const key = resolveMap[fieldId];
+  let key;
+  if (/\./.test(fieldId)) {
+    key = fieldId;
+  } else {
+    key = resolveMap[fieldId];
+  }
   if (key) {
-    let field = getTransformedField(fieldId, fields[fieldId]);
-    app[key] = field;
+    let field = getTransformedField(fieldId, objectPath.get(fields, fieldId));
```

Comment:
The way I see it is that we might want to validate the whole of the container object, instead of the individual fields that make up the container settings. 
The object path allows us to trace back to which "top-level" object the field belongs to. Imagine if you are supplying docker.container.ports but image is empty (or some combination of values that is invalid).